controllers/controllers.js

```diff
@@ -42,16 +42,57 @@ exports.getArticleByID = (req, res, next) => {
 exports.getCommentsByArticleID = (req, res, next) => {
     const {article_id} = req.params
     const promises = [selectArticleByID(article_id)]
-
+    
     Promise.all(promises).then(() => {
         selectCommentsByArticle(article_id).then((comments) => {
             res.status(200).send({comments})
         })
-
+        
     }).catch((err) => {
         next(err)
     })
-
+    
 }
 
+exports.postCommentsByArticleID = (req, res, next) => {
+    const {article_id} = req.params
+    const promises = [selectArticleByID(article_id)]
+    
+    Promise.all(promises).then(() => {
+        if (!req.body.hasOwnProperty('username') || !req.body.hasOwnProperty('body')){
+            return Promise.reject({status: 400, msg: 'Bad Request'})
+        }
+        
+        const username = req.body.username
+        const body = req.body.body
+        return insertCommentByArticleId(article_id, username, body).then((comment) => {
+            res.status(201).send({comment: comment[0]})
+            
+        })
+    }).catch((err) => {
+        next(err)
+    })
+    
+    
+}
 
+exports.patchVoteByArticleID = (req, res, next) => {
+    const {article_id} = req.params
+    const promises = [selectArticleByID(article_id)]
+    
+    
+    Promise.all(promises).then(() => {
+        if (!req.body.hasOwnProperty('inc_votes')){
+            return Promise.reject({status: 400, msg: 'Bad Request'})
+        }
+        
+        const votes = req.body.inc_votes
+        return insertVoteByArticleID(article_id, votes).then((article) => {
+            res.status(200).send({article: article[0]})
+    
+    })
+    }).catch((err) => {
+        next(err)
+    }) 
+    
```

Comment:
When there's only one pending element in the Promise.all() array, it doesn't need to use Promise.all() Can just call the promise rather than putting it into Promise.all()